projects/ion/src/lib/tooltip/tooltip.component.ts

```diff
@@ -1,17 +1,52 @@
-import { Component, TemplateRef } from '@angular/core';
+import {
+  AfterViewChecked,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  TemplateRef,
+  ViewChild,
+} from '@angular/core';
 import { TooltipColorScheme, TooltipPosition } from '../core/types';
+import { TooltipService } from './tooltip.service';
+
+const PADDING = 16;
+export const TOOLTIP_MAX_WIDTH = 208 + PADDING;
 
 @Component({
   selector: 'ion-tooltip',
   templateUrl: './tooltip.component.html',
   styleUrls: ['./tooltip.component.scss'],
 })
-export class IonTooltipComponent {
+export class IonTooltipComponent implements AfterViewChecked {
+  @ViewChild('tooltip', { static: true }) tooltip: ElementRef;
+
   ionTooltipTitle: string;
   ionTooltipTemplateRef: TemplateRef<void>;
   ionTooltipColorScheme: TooltipColorScheme = 'dark';
   ionTooltipPosition: TooltipPosition = TooltipPosition.DEFAULT;
   ionTooltipVisible = false;
   left = 0;
   top = 0;
+
+  constructor(
+    private cdr: ChangeDetectorRef,
+    private tooltipService: TooltipService
+  ) {}
+
+  ngAfterViewChecked(): void {
+    this.repositionTooltip();
+    this.cdr.detectChanges();
+  }
+
+  private repositionTooltip(): void {
+    this.sendCoordinates();
+    this.tooltipService.screenSize = document.body;
+    this.ionTooltipPosition = this.tooltipService.getNewPosition();
+    this.tooltipService.reposition.next();
```

Comment:
```suggestion
    this.tooltipService.emitReposition();
```
Try to make this a method, to be more suggestive of what it is actually doing.